pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -0,0 +1,789 @@
+}
+
+func (d *stateData) Clone() framework.StateData {
+	return d
+}
+
+func (d *stateData) updateClaimStatus(ctx context.Context, clientset kubernetes.Interface, index int, claim *resourcev1alpha1.ResourceClaim) error {
+	// TODO (?): replace with patch operation. Beware that patching must only succeed if the
+	// object has not been modified in parallel by someone else.
+	claim, err := clientset.ResourceV1alpha1().ResourceClaims(claim.Namespace).UpdateStatus(ctx, claim, metav1.UpdateOptions{})
+	// TODO: metric for update results, with the operation ("set selected
+	// node", "set PotentialNodes", etc.) as one dimension.
+	if err != nil {
+		return fmt.Errorf("update resource claim: %w", err)
+	}
+
+	// Remember the new instance. This is relevant when the plugin must
+	// update the same claim multiple times (for example, first set
+	// PotentialNodes, then SelectedNode), because otherwise the second
+	// update would fail with a "was modified" error.
+	d.claims[index] = claim
+
+	return nil
+}
+
+// initializePodScheduling can be called concurrently. It returns an existing PodScheduling
+// object if there is one already, retrieves one if not, or as a last resort creates
+// one from scratch.
+func (d *stateData) initializePodScheduling(ctx context.Context, pod *v1.Pod, podSchedulingLister resourcev1alpha1listers.PodSchedulingLister) (*resourcev1alpha1.PodScheduling, error) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	if d.podScheduling != nil {
+		return d.podScheduling, nil
+	}
+
+	podScheduling, err := podSchedulingLister.PodSchedulings(pod.Namespace).Get(pod.Name)
+	switch {
+	case apierrors.IsNotFound(err):
+		controller := true
+		podScheduling = &resourcev1alpha1.PodScheduling{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: "v1",
+						Kind:       "Pod",
+						Name:       pod.Name,
+						UID:        pod.UID,
+						Controller: &controller,
+					},
+				},
+			},
+		}
+		err = nil
+	case err != nil:
+		return nil, err
+	default:
+		// We have an object, but it might be obsolete.
+		if !metav1.IsControlledBy(podScheduling, pod) {
+			return nil, fmt.Errorf("PodScheduling object with UID %s is not owned by Pod %s/%s", podScheduling.UID, pod.Namespace, pod.Name)
+		}
+	}
+	d.podScheduling = podScheduling
+	return podScheduling, err
+}
+
+// publishPodScheduling creates or updates the PodScheduling object.
+func (d *stateData) publishPodScheduling(ctx context.Context, clientset kubernetes.Interface, podScheduling *resourcev1alpha1.PodScheduling) error {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	var err error
+	logger := klog.FromContext(ctx)
+	msg := "updating PodScheduling"
+	if podScheduling.UID == "" {
+		msg = "creating PodScheduling"
+	}
+	if loggerV := logger.V(6); loggerV.Enabled() {
+		// At a high enough log level, dump the entire object.
+		logger.V(5).Info(msg, "podscheduling", podScheduling)
```

Comment:
```suggestion
		logger.V(5).Info(msg, "podscheduling", klog.KObj(podScheduling), "podschedulingDump", podScheduling)
```

for easier filtering.

Reply:
Changed. Also, it should have logged as loggerV = logger.V(6) - minor cosmetic difference, results in v=6 in JSON output.